internal/controller/manager.go

```diff
@@ -121,8 +124,26 @@ func registerControllersWithManager(mgr ctrl.Manager, config *ManagerConfig) err
 	}
 
 	// Add secret reconciler to the manager
-	return secret.NewReconciler(
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
+	defer cancel()
+	if err = secret.NewReconciler(
 		mgr,
 		config.SecretControllerConfig,
-	).RegisterWithManager(ctx, mgr)
+	).RegisterWithManager(ctx, mgr); err != nil {
+		return err
+	}
+
+	// Add sentinel webhook to the manager
+	if config.SentinelWebhookConfig.Enabled {
+		var sentinelWebhook *sentinel.Handler
+		if sentinelWebhook, err = sentinel.NewHandler(
+			mgr,
+			config.SentinelWebhookConfig,
+		); err != nil {
+			return err
+		}
+		return sentinelWebhook.RegisterWithManager(mgr)
+	}
```

Comment:
It might be an overkill to have a separate `manager.go` or more correctly `register.go` just for webhooks, simply to aggregate registration of all webhooks before registering with the manager, but I'm not opposed to the idea.

In any case, I will rename the function `registerControllersWithManager ()` to `registerControllersAndWebhooksWithManager()` instead, to tell us the correct intent of the function.

In our previous refactor of druid, we brought the controller-manager code under `controllers`, and then had each controller package under that. But now that we have webhooks, it might make sense to bring the controller-manager code into a separate package, and then register controllers and webhooks for it. In that case, `internal/controller` and `internal/webhook` can have their own aggregators for registering all the controllers/webhooks under them, and the controller-manager can simply make two calls, one to register all controllers, and the other to register all webhooks. @unmarshall WDYT?

Reply:
The rename to `registerControllersAndWebhooksWithManager()` might be good enough for now.

But a `manager` package at `internal/manager` which follows your suggestion, is a better way to do this in my opinion.

Reply:
Addressed in https://github.com/gardener/etcd-druid/pull/777/commits/c0895d233daee0e4aba48a2740cba1b90161adfd

We have now created a new package `internal/manager`, and all manager code goes there. It invokes functions from `internal/controller/*` and `internal/webhook/*` packages. I think this is sufficient, and we don't need wrapper functions for controllers and webhooks again for manager to use. WDYT?

Also, the function `CreateManagerWithControllersAndWebhooks()` is now renamed to `InitializeManager()`, which makes it short and sweet :) . Also, `registerControllersAndWebhooksWithManager()` is split into `registerControllers()` and `registerWebhooks()` for better modularity and better naming too.

Reply:
These changes are great! Thanks for introducing `internal/manager`.